src/main/java/gregtech/api/sound/GTSoundManager.java

```diff
@@ -0,0 +1,49 @@
+package gregtech.api.sound;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.audio.ISound;
+import net.minecraft.client.audio.PositionedSoundRecord;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@SideOnly(Side.CLIENT)
+@Mod.EventBusSubscriber(Side.CLIENT)
+public class GTSoundManager {
+
+    private static final Map<BlockPos, ISound> soundMap = new HashMap<>();
+
+    public static ISound startTileSound(ResourceLocation soundName, float volume, BlockPos pos) {
+        ISound s = soundMap.get(pos);
+        if (s == null || !Minecraft.getMinecraft().getSoundHandler().isSoundPlaying(s)) {
+            s = new PositionedSoundRecord(soundName, SoundCategory.BLOCKS, volume, 1.0F,
+                    true, 0, ISound.AttenuationType.LINEAR, pos.getX() + 0.5F, pos.getY() + 0.5F, pos.getZ() + 0.5F) {
+                @Override
+                public float getVolume() {
+                    if (this.sound == null) {
+                        this.createAccessor(Minecraft.getMinecraft().getSoundHandler());
+                    }
+                    return super.getVolume();
+                }
+            };
```

Comment:
Needed? AFAICS `createAccessor` gets called after `playSound` is called. `sound` should never be null, though can be `SoundHandler#MISSING_SOUND`.

Reply:
Ah, thinking about it more, I think this was used in Mekanism because of the way they handled partial muffling with Muffler Upgrades. Should be safe to remove for us